### PR TITLE
[PIL-1082] - Add audit events for RFM

### PR DESCRIPTION
**Pull request by jassalrichy**
Added "Pillar2ReplaceFilingMember" audit event

### CODE REVIEW

File: app/services/audit/AuditService.scala [hunk 1]
```diff
@@ -165,4 +166,33 @@ class AuditService @Inject() (
...
+  def auditReplaceFilingMember(nfmDetail: NewFilingMemberDetail)(implicit hc: HeaderCarrier): Future[AuditResult] =
+    auditConnector.sendExtendedEvent(
+      RfmAuditEvent(
+        nfmDetail.plrReference,
+        nfmDetail.securityAnswerRegistrationDate,
+        nfmDetail.corporatePosition,
+        nfmDetail.ukBased match {
+          case Some(value) => Some(value)
+          case _           => None
+        },
```
[the remark this answers is not in the document]
Comment (jassalrichy) [reply app/services/audit/AuditService.scala:179]: Well spotted! - that was a test to see if anyone noticed ... actually just a moment of madness

File: app/services/audit/AuditService.scala [hunk 2]
```diff
@@ -165,4 +166,33 @@ class AuditService @Inject() (
...
+        nfmDetail.registeredAddress,
+        nfmDetail.primaryContactName,
+        nfmDetail.primaryContactEmail,
+        nfmDetail.primaryContactPhonePreference,
+        nfmDetail.primaryContactPhoneNumber,
+        nfmDetail.addSecondaryContact,
+        nfmDetail.secondaryContactInformation.map(scInfo => scInfo.name),
+        nfmDetail.secondaryContactInformation.map(scInfo => scInfo.emailAddress),
+        nfmDetail.secondaryContactInformation.map(scInfo => scInfo.telephone.isDefined),
+        nfmDetail.secondaryContactInformation match {
```
[the remark this answers is not in the document]
Comment (jassalrichy) [reply app/services/audit/AuditService.scala:190]: Yep, changed to flatMap -> map

File: app/controllers/rfm/RfmContactCheckYourAnswersController.scala [hunk 3]
```diff
@@ -114,14 +115,18 @@ class RfmContactCheckYourAnswersController @Inject() (
...
-               subscriptionService.allocateEnrolment(groupId = groupId, plrReference = newFilingMemberInformation.plrReference, upeEnrolmentInfo)
-             )
-        _                   <- liftF(userAnswersConnectors.remove(request.userId))
+        _ <- liftF(subscriptionService.allocateEnrolment(groupId = groupId, plrReference = newFilingMemberInformation.plrReference, upeEnrolmentInfo))
+        _ <- liftF(userAnswersConnectors.remove(request.userId))
         optionalSessionData <- liftF(sessionRepository.get(request.userAnswers.id))
         sessionData = optionalSessionData.getOrElse(UserAnswers(request.userId))
         updatedSessionData <- liftF(Future.fromTry(sessionData.set(PlrReferencePage, newFilingMemberInformation.plrReference)))
         _                  <- liftF(sessionRepository.set(updatedSessionData))
+        isNewNfmNonUkBased = (newFilingMemberInformation.corporatePosition, newFilingMemberInformation.ukBased) match {
```
[the remark this answers is not in the document]
Comment (jassalrichy) [reply app/controllers/rfm/RfmContactCheckYourAnswersController.scala]: Done

File: app/controllers/rfm/RfmContactCheckYourAnswersController.scala [hunk 4]
```diff
@@ -114,14 +115,18 @@ class RfmContactCheckYourAnswersController @Inject() (
                               )
                             )
         groupId <- fromOption[Future](request.groupId)
-        _ <- liftF(
-               subscriptionService.allocateEnrolment(groupId = groupId, plrReference = newFilingMemberInformation.plrReference, upeEnrolmentInfo)
-             )
-        _                   <- liftF(userAnswersConnectors.remove(request.userId))
+        _ <- liftF(subscriptionService.allocateEnrolment(groupId = groupId, plrReference = newFilingMemberInformation.plrReference, upeEnrolmentInfo))
```
[the remark this answers is not in the document]
Comment (jassalrichy) [reply app/controllers/rfm/RfmContactCheckYourAnswersController.scala:116]: Slightly refactored.

**Review by javed-iqbal1 — APPROVED** [review]

**Pull request merged**